modP2pImpl/src/org/aion/p2p/impl/comm/NodeMgr.java

```diff
@@ -358,49 +361,67 @@ public synchronized void moveOutboundToActive(int _nodeIdHash, String _shortId)
     }
 
     void timeoutInbound() {
-        Iterator<Integer> inboundIt = inboundNodes.keySet().iterator();
-        while (inboundIt.hasNext()) {
-            int key = inboundIt.next();
-            INode node = inboundNodes.get(key);
-            if (System.currentTimeMillis() - node.getTimestamp() > TIMEOUT_INBOUND_NODES) {
-                p2pMgr.closeSocket(node.getChannel(), "inbound-timeout ip=" + node.getIpStr());
-                inboundIt.remove();
+        try {
+            Iterator<Map.Entry<Integer, INode>> it = inboundNodes.entrySet().iterator();
+            while (it.hasNext()) {
+                Map.Entry<Integer, INode> entry = it.next();
+                if (System.currentTimeMillis() - entry.getValue().getTimestamp()
+                    > TIMEOUT_INBOUND_NODES) {
+                    p2pMgr.closeSocket(entry.getValue().getChannel(),
+                        "inbound-timeout ip=" + entry.getValue().getIpStr());
+                    it.remove();
+                }
             }
+        } catch (IllegalStateException e) {
+            e.printStackTrace();
+            p2pLOG.debug("timeoutInbound IllegalStateException ", e.getMessage());
         }
     }
 
     private void timeoutActive() {
-        long now = System.currentTimeMillis();
 
-        OptionalDouble average = activeNodes.values().stream()
+        long now = System.currentTimeMillis();
+        OptionalDouble average = activeNodes.values().parallelStream()
```

Comment:
Whats the purpose of using a ``parallelStream()`` here? My assumption is that the overhead of threading would be pretty heavy.

Reply:
reverted it back  https://github.com/aionnetwork/aion/pull/539/commits/fa5399283d7e3795f8d967454460894f3b8d27dd
http://gee.cs.oswego.edu/dl/html/StreamParallelGuidance.html
only the number of elements more than 10K than worth to use it.